src/lib/rust.ts

```diff
@@ -300,13 +305,36 @@ export const ensureCargoDependency = async ({
     });
 
   try {
-    if (process.env.CI === "true") {
+    if (
+      force ||
+      ![MARINE_CARGO_DEPENDENCY, MREPL_CARGO_DEPENDENCY].includes(name)
+    ) {
+      throw new Error("Install using cargo");
+    }
+
+    if (["darwin", "linux"].includes(platform()) && arch() === "x64") {
       await downloadPrebuiltCargoDependencies(name, version, dependencyPath);
     } else {
-      throw new Error("Not in CI");
+      throw new Error("Unsupported platform. Use cargo");
+    }
+
+    try {
+      // check binary is working
+      const helpText = await execPromise({
+        command: join(dependencyPath, "bin", name),
+        args: ["--help"],
+      });
+
+      if (!helpText.includes(version)) {
+        throw new Error("Version mismatch");
+      }
+    } catch {
+      await rm(dependencyPath, { recursive: true, force: true });
+      throw new Error("Not working. Fallback to normal installation");
```

Comment:
What is not working?

Reply:
I understand your confusion, I will think about how can I avoid using these errors so it's more clear what is happening